localstack/services/s3/s3_listener.py

```diff
@@ -1059,8 +1059,11 @@ def handle_put_bucket_notification(bucket, data):
         configs = config if isinstance(config, list) else [config] if config else []
         for config in configs:
             events = config.get("Event")
+            if isinstance(events, dict) and "@xmlns" in events:
```

Comment:
Thanks for this PR @benesch ! 👍 

Can we please add a small test to cover this new functionality? I think we could create a new unit test in `tests/unit/test_s3.py` and simply pass in a random bucket name and XML payload to the `handle_put_bucket_notification(..)` function, and then assert that the correct notification entries are added to `S3_NOTIFICATIONS`.

Reply:
Thanks for the very quick review @whummer! I added the test as requested. I also refactored the code into a function that can be applied to any object returned by xmltodict.parse, which is quite a bit more general purpose and works for the `Id` and `Filter` elements too.

Reply:
Looks great, thanks for updating the PR 👍